strawberry/quart/views.py

```diff
@@ -123,15 +178,58 @@ async def create_streaming_response(
         )
 
     def is_websocket_request(self, request: Request) -> TypeGuard[Request]:
-        return False
+        if has_websocket_context():
+            return True
+
+        # Check if the request is a WebSocket upgrade request
+        connection = request.headers.get("Connection", "").lower()
+        upgrade = request.headers.get("Upgrade", "").lower()
+
+        return "upgrade" in connection and "websocket" in upgrade
 
     async def pick_websocket_subprotocol(self, request: Request) -> Optional[str]:
-        raise NotImplementedError
+        # Get the requested protocols
+        protocols_header = websocket.headers.get("Sec-WebSocket-Protocol", "")
+        if not protocols_header:
+            return None
+
+        # Find the first matching protocol
+        requested_protocols = [p.strip() for p in protocols_header.split(",")]
+        for protocol in requested_protocols:
+            if protocol in self.subscription_protocols:
+                return protocol
+
+        return None
 
     async def create_websocket_response(
         self, request: Request, subprotocol: Optional[str]
     ) -> Response:
-        raise NotImplementedError
+        if subprotocol:
+            # Set the WebSocket protocol if specified
+            await websocket.accept(subprotocol=subprotocol)
+        else:
+            await websocket.accept()
```

Comment:
The `.accept` methods `subprotocol` argument accepts `Optional[str]`. Unless I'm missing something, we don't have to perform a check here.

```suggestion
        await websocket.accept(subprotocol=subprotocol)
```